From 452cd46bccc074ce91a2f482b7fb9d3f89f0aede Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Wed, 26 Nov 2014 11:25:53 +0100
Subject: [PATCH 2/2] Add support for the Ericsson F5521gw modem

The modem seems to be a bit picky about its serial configration.

See http://logs.nslu2-linux.org/livelogs/ofono/ofono.20110712.txt for more details.

Signed-off-by: Simon Busch <morphis@gravedo.de>
Upstream-Status: Pending
---
 ofono/gatchat/gattty.c | 2 ++
 ofono/plugins/mbm.c    | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/ofono/gatchat/gattty.c b/ofono/gatchat/gattty.c
index 3e447e29..f3eb2f00 100644
--- a/ofono/gatchat/gattty.c
+++ b/ofono/gatchat/gattty.c
@@ -236,6 +236,8 @@ static int open_device(const char *tty, GHashTable *options)
 
 	tcflush(fd, TCIOFLUSH);
 	tcsetattr(fd, TCSANOW, &ti);
+	tcflush(fd, TCIOFLUSH);
+	tcflush(fd, TCIOFLUSH);
 
 	return fd;
 }
diff --git a/ofono/plugins/mbm.c b/ofono/plugins/mbm.c
index 7d6f92ab..f1af6b91 100644
--- a/ofono/plugins/mbm.c
+++ b/ofono/plugins/mbm.c
@@ -287,6 +287,10 @@ static GAtChat *create_port(const char *device)
 		return NULL;
 
 	g_hash_table_insert(options, "Baud", "115200");
+	g_hash_table_insert(options, "Parity", "none");
+	g_hash_table_insert(options, "StopBits", "1");
+	g_hash_table_insert(options, "DataBits", "8");
+	g_hash_table_insert(options, "RtsCts", "on");
 
 	channel = g_at_tty_open(device, options);
 
-- 
2.17.0

